exercises.py

```diff
@@ -1,13 +1,394 @@
+d
+
+# 4. What does spam[-1] evaluate to?
+d
+
+# 5. What does spam[:2] evaluate to?
+['a', 'b']
+
+# For the following three questions, let’s say bacon contains the list [3.14, 'cat', 11, 'cat', True].
+# 6. What does bacon.index('cat') evaluate to?
+1
+
+# 7. What does bacon.append(99) make the list value in bacon look like?
+[3.14, 'cat', 11, 'cat', True, 99]
+
+# 8. What does bacon.remove('cat') make the list value in bacon look like?
+[3.14, 11, 'cat', True, 99]
+
+# 9. What are the operators for list concatenation and list replication?
++ for concatenation, and * for replication
+
+# 10. What is the difference between the append() and insert() list methods?
+.append() adds something to the end of a list, and .insert ( , ) adds something to an indexed spot in the list
+
+# 11. What are two ways to remove values from a list?
+del list_name[index value] and list_name.remove()
+
+# 12. Name a few ways that list values are similar to string values.
+They both start and end with something string values use "", list values use []
+they are both "lists" of single text characters
+
+# 13. What is the difference between lists and tuples?
+tuples use () instead of []
+tuples contain 2+ items of related data (thing1, thing2, thing 3)
+tuples are immutable, and cannot have their values modified, appended, or removed
+use tuples when you dont intend for the sequence of values to be modified
+
+# 14. How do you type the tuple value that has just the integer value 42 in it?
+(42,)
+
+# 15. How can you get the tuple form of a list value? How can you get the list form of a tuple value?
+tuple([item1, item2, item3])  to get the tuple form of a list value
+list((item2, item2, item3))  to gdt the list form of a tuple value
+
+# 16. Variables that “contain” list values don’t actually contain lists directly. What do they contain instead?
+they contain a reference to a list 
+
+# 17. What is the difference between copy.copy() and copy.deepcopy()?
+copy.copy() makes a duplicate copy of a mutable value like a list or dictionary
+copy.deepcopy() will also copy inner lists that might be inside the list, instead of just the list itself
+
+#############################################################################################
+
+
+
+
+
+
+
 # Chapter 5
 
+# 1. What does the code for an empty dictionary look like?
+{:}
+
+# 2. What does a dictionary value with a key 'foo' and a value 42 look like?
+{foo : 42}
+
+# 3. What is the main difference between a dictionary and a list?
+a dictionary has a key that corresponds to a value  {"key1" : value1, "key2" : value 2, "key3": value 3}
+
+# 4. What happens if you try to access spam['foo'] if spam is {'bar': 100}?
+you would get an error because that key isnt defined 
+KeyError: "foo"
+
+# 5. If a dictionary is stored in spam, what is the difference between the expressions 'cat' in spam and 'cat' in spam.keys()?
+expressing cat in dictionary spam would yield the value corresponding to key "cat"
```

Comment:
`spam['cat']` or `spam.get('cat') would yield the value corresponding to the key `'cat'`

Reply:
`cat in spam` would indicate whether 'cat' appears among the keys of the dictionary.